resources/setup_emulator.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+set -e
+
+kill_all_emus() {
+    for emu_device in $(adb devices -l |grep 'device product:' |cut -d' ' -f1); do
+        echo "Killing emulator: $emu_device"
+        adb -s $emu_device emu kill
+    done
+}
+
+boot_finished_status="stopped"
+attempt=0
+timeout=300
+
+echo 'Creating emulator'
+echo 'n' | $ANDROID_HOME/tools/bin/avdmanager create avd -n test-android -f -k "system-images;android-27;google_apis;x86" 
+
+echo 'loading emulator'
+$ANDROID_HOME/emulator/emulator -avd test-android -no-boot-anim -no-window -noaudio & 
+
+while true; do
+boot_current_status=`adb -e shell getprop init.svc.bootanim &`
+if [ "$boot_current_status" = "$boot_finished_status" ]; 
+then 
+    echo "emulator is ready"
+    break
+else 
+    let "attempt += 1"
+    echo "$attempt sec, waiting more for emulator to start"
+    if [ $attempt -gt $timeout ]; then
+      echo "Timeout ($timeout seconds) reached; failed to start emulator"
+      exit 1
+    fi
+    sleep 1
+fi    
+done
+
+EXPECTED_DEVICE_NAME='emulator-5554'
```

Comment:
Does this number ever change?

Reply:
Every first emulator will have this name 'emulator-5554' by default and we'll be closing/killing all existing emulators(if any) before creating new one(it will also improve execution speed overall)